src/main/java/com/google/firebase/auth/FirebaseAuth.java

```diff
@@ -244,7 +283,40 @@ public FirebaseToken call() throws Exception {
    *     unsuccessfully with the failure Exception.
    */
   public ApiFuture<FirebaseToken> verifyIdTokenAsync(final String token) {
-    return new TaskToApiFuture<>(verifyIdToken(token));
+    return verifyIdTokenAsync(token, false);
+  }
+
+  /**
+   * Parses and verifies a Firebase ID Token and if requested, checks whether it was revoked.
+   *
+   * <p>A Firebase application can identify itself to a trusted backend server by sending its
+   * Firebase ID Token (accessible via the getToken API in the Firebase Authentication client) with
+   * its request.
+   *
+   * <p>The backend server can then use the verifyIdToken() method to verify the token is valid,
+   * meaning: the token is properly signed, has not expired, and it was issued for the project
+   * associated with this FirebaseAuth instance (which by default is extracted from your service
+   * account)
+   * 
+   * <p>If a request was made to check revoked, the issued-at property of the token (like all 
```

Comment:
Way too much implementation details here. Just mention something like: "If checkRevoked is true, additionally checks if the token has been revoked."